src/components/member/PasswordSetting.js

```diff
@@ -0,0 +1,228 @@
+import { makeStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import { Button, TextField } from '@material-ui/core';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { useMutation } from '../../config/queryClient';
+import {
+  CONFIRM_CHANGE_PASSWORD_BUTTON_ID,
+  CONFIRM_RESET_PASSWORD_BUTTON_ID,
+  USER_CONFIRM_PASSWORD_INPUT_ID,
+  USER_CURRENT_PASSWORD_INPUT_ID,
+  USER_NEW_PASSWORD_INPUT_ID,
+} from '../../config/selectors';
+import {
+  newPasswordValidator,
+  passwordValidator,
+  strengthValidator,
+} from '../../utils/validation';
+
+const useStyles = makeStyles(() => ({
+  mainContainer: {
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    margin: '12px 0px',
+  },
+  changePasswordContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    margin: '12px 0px',
+  },
+  buttonItem: {
+    display: 'flex',
+    padding: '12px',
+    flexDirection: 'column',
+    justifyContent: 'flex-end',
+    alignItems: 'flex-start',
+  },
+  note: {
+    fontSize: '12px',
+    margin: '4px 0 2px',
+  },
+}));
+
+const PasswordSetting = ({ user }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [currentPasswordError, setCurrentPasswordError] = useState(null);
+  const [newPasswordError, setNewPasswordError] = useState(null);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(null);
+  const { mutate: onUpdatePassword } = useMutation(
+    MUTATION_KEYS.UPDATE_PASSWORD,
+  );
+
+  const userId = user.get('id');
+
+  const onClose = () => {
+    setCurrentPassword('');
+    setNewPassword('');
+    setConfirmPassword('');
+  };
+
+  const handleChangePassword = () => {
+    const checkingCurrentPassword = passwordValidator(currentPassword);
+    const checkingNewPassword = passwordValidator(newPassword);
+    const checkingConfirmPassword = passwordValidator(confirmPassword);
+    // verify empty fields
+    if (
+      user.get('password') &&
+      (checkingCurrentPassword ||
+        checkingNewPassword ||
+        checkingConfirmPassword)
+    ) {
+      setCurrentPasswordError(checkingCurrentPassword);
+      setNewPasswordError(checkingNewPassword);
+      setConfirmPasswordError(checkingConfirmPassword);
+    } else {
+      const checkingPasswordUpdate = newPasswordValidator(
+        currentPassword,
+        newPassword,
+        confirmPassword,
+      );
+      if (checkingPasswordUpdate) {
+        toast.error(checkingPasswordUpdate);
+      } else {
+        const checkingStrength = strengthValidator(newPassword);
+        if (checkingStrength) {
+          toast.error(checkingStrength);
+        } else {
+          onUpdatePassword({
+            id: userId,
```

Comment:
Do you really need this id? I think you can only change your own password.

Reply:
It is necessary because the task manager checks if the member is the same as the one trying to perform the update operation

Reply:
Indeed, but since I've commented in your other PR that you should have a dedicated endpoint for setting a password, you won't need the id anymore (since the endpoint won't depend on the user id)!